ci/build-check.sh

```diff
@@ -10,16 +10,3 @@ ${dn}/build.sh
 # support parallel runs right now
 /usr/bin/make check
 make install
-
-# And now a clang build to find unused variables, but only run on Fedora because
-# the CentOS version is ancient anyway and doesn't support all the flags that
-# might get passed to it.
-id=$(. /etc/os-release && echo $ID)
-if [ "$id" == fedora ]; then
-    git clean -dfx
```

Comment:
It turns out that libdnf is not ready to build with `clang`, and we'd subtly been avoiding that by not doing a `git submodule foreach git clean -dfx` like ostree does.  For now let's just back this out while I try to unwind that.